components/livekit/tests/livekit-component.spec.ts

```diff
@@ -0,0 +1,643 @@
+import { ILoggerComponent } from '@well-known-components/interfaces'
+import { createLoggerMockedComponent } from '@dcl/core-commons'
+import { createLivekitComponent } from '../src/component'
+import { ILivekitComponent, LivekitSettings, IngressInput } from '../src/types'
+import { LivekitIngressNotFoundError, LivekitWebhookVerificationError } from '../src/errors'
+
+// Mock livekit-server-sdk
+jest.mock('livekit-server-sdk', () => {
+  const mockToJwt = jest.fn().mockResolvedValue('mock-jwt-token')
+  const mockAddGrant = jest.fn()
+
+  return {
+    AccessToken: jest.fn().mockImplementation(() => ({
+      addGrant: mockAddGrant,
+      toJwt: mockToJwt
+    })),
+    RoomServiceClient: jest.fn().mockImplementation(() => ({
+      listRooms: jest.fn(),
+      createRoom: jest.fn(),
+      deleteRoom: jest.fn(),
+      updateRoomMetadata: jest.fn(),
+      listParticipants: jest.fn(),
+      removeParticipant: jest.fn(),
+      updateParticipant: jest.fn()
+    })),
+    IngressClient: jest.fn().mockImplementation(() => ({
+      listIngress: jest.fn(),
+      createIngress: jest.fn(),
+      deleteIngress: jest.fn()
+    })),
+    WebhookReceiver: jest.fn().mockImplementation(() => ({
+      receive: jest.fn()
+    })),
+    IngressInput: {
+      RTMP_INPUT: 0,
+      WHIP_INPUT: 1,
+      URL_INPUT: 2
+    },
+    TrackSource: {
+      CAMERA: 0,
+      MICROPHONE: 1,
+      SCREEN_SHARE: 2,
+      SCREEN_SHARE_AUDIO: 3
+    }
+  }
+})
+
+let logs: ILoggerComponent
+let component: ILivekitComponent
+let settings: LivekitSettings
+let mockRoomClient: any
+let mockIngressClient: any
+let mockWebhookReceiver: any
+
+beforeEach(async () => {
+  logs = createLoggerMockedComponent()
+  settings = {
+    host: 'wss://livekit.example.com',
+    apiKey: 'test-api-key',
+    secret: 'test-api-secret'
+  }
+
+  // Get the mock instances
+  const { RoomServiceClient, IngressClient, WebhookReceiver } = require('livekit-server-sdk')
+
+  component = await createLivekitComponent({ logs }, { settings })
+
+  // Get the mock instances that were created
+  mockRoomClient = RoomServiceClient.mock.results[0].value
+  mockIngressClient = IngressClient.mock.results[0].value
+  mockWebhookReceiver = WebhookReceiver.mock.results[0].value
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('when creating the livekit component', () => {
+  describe('and the host URL does not have wss:// prefix', () => {
+    it('should normalize the host URL by adding wss:// prefix', async () => {
+      const settingsWithoutProtocol = {
```

Comment:
Instead of creating a `settingsWithoutProtocol` variable, WDYT about modfying the settings property in a `beforeEach` below the `and the host URL does not have wss:// prefix` describe?